src/main/java/org/folio/dataexp/service/export/strategies/ld/AbstractLinkedDataExportStrategy.java

```diff
@@ -0,0 +1,245 @@
+import org.folio.dataexp.repository.JobProfileEntityRepository;
+import org.folio.dataexp.service.export.LocalStorageWriter;
+import org.folio.dataexp.service.export.strategies.ExportStrategy;
+import org.folio.dataexp.service.export.strategies.ExportStrategyStatistic;
+import org.folio.dataexp.service.export.strategies.ExportedMarcListener;
+import org.folio.dataexp.service.logs.ErrorLogService;
+import org.folio.dataexp.util.S3FilePathUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.PageRequest;
+
+/**
+ * Abstract base class for Linked Data export strategies, providing common logic
+ * that all implementations can use while providing their own retrievel mechanism.
+ */
+@Log4j2
+@Getter
+public abstract class AbstractLinkedDataExportStrategy implements ExportStrategy {
+  protected int exportIdsBatch;
+  protected String exportTmpStorage;
+
+  private ExportIdEntityRepository exportIdEntityRepository;
+  private JobProfileEntityRepository jobProfileEntityRepository;
+  private LinkedDataConverter linkedDataConverter;
+
+  protected ErrorLogService errorLogService;
+
+  @Value("#{ T(Integer).parseInt('${application.export-ids-batch}')}")
+  protected void setExportIdsBatch(int exportIdsBatch) {
+    this.exportIdsBatch = exportIdsBatch;
+  }
+
+  @Value("${application.export-tmp-storage}")
+  protected void setExportTmpStorage(String exportTmpStorage) {
+    this.exportTmpStorage = exportTmpStorage;
+  }
+
+  @Override
+  public ExportStrategyStatistic saveOutputToLocalStorage(
+      JobExecutionExportFilesEntity exportFilesEntity,
+      ExportRequest exportRequest,
+      ExportedMarcListener exportedMarcListener
+  ) {
+    var exportStatistic = new ExportStrategyStatistic(exportedMarcListener);
+    var localStorageWriter = createLocalStorageWriter(exportFilesEntity);
+    processSlices(exportFilesEntity, exportStatistic, localStorageWriter);
+    try {
+      localStorageWriter.close();
+    } catch (Exception e) {
+      log.error(
+          "saveOutputToLocalStorage: Error saving file {} to local storage for job execution {}",
+          exportFilesEntity.getFileLocation(),
+          exportFilesEntity.getJobExecutionId()
+      );
+      exportStatistic.setDuplicatedSrs(0);
+      exportStatistic.removeExported();
+      long countFailed = exportIdEntityRepository.countExportIds(
+          exportFilesEntity.getJobExecutionId(),
+          exportFilesEntity.getFromId(),
+          exportFilesEntity.getToId()
+      );
+      exportStatistic.setFailed((int) countFailed);
+    }
+    return exportStatistic;
+  }
+
+  @Override
+  public void setStatusBaseExportStatistic(
+      JobExecutionExportFilesEntity exportFilesEntity,
+      ExportStrategyStatistic exportStatistic
+  ) {
+    if (exportStatistic.getFailed() == 0 && exportStatistic.getExported() > 0) {
```

Comment:
`exportStatistic.getFailed() == 0 && exportStatistic.getExported()` - I would propose to collapse this condition and also two conditions bellow to implement separate boolean methods: isCompleted, isCompletedWithErrors and isFailed. This can improve codes readability IMO. Does it make sense?